src/transformers/generation_utils.py

```diff
@@ -580,6 +586,10 @@ def _get_logits_processor(
             processors.append(MinLengthLogitsProcessor(min_length, eos_token_id))
         if prefix_allowed_tokens_fn is not None:
             processors.append(PrefixConstrainedLogitsProcessor(prefix_allowed_tokens_fn, num_beams))
+        if forced_bos_token_id is not None and num_beams > 1:
```

Comment:
I'd say we keep it as general as possible and also allow it for `num_beams == 0` (there could be cases where people want to force the 2nd token